app/src/main/java/com/hedvig/app/feature/marketpicker/LanguageAdapter.kt

```diff
@@ -0,0 +1,123 @@
+package com.hedvig.app.feature.marketpicker
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.RadioButton
+import android.widget.TextView
+import androidx.dynamicanimation.animation.SpringAnimation
+import androidx.dynamicanimation.animation.SpringForce
+import androidx.recyclerview.widget.RecyclerView
+import com.hedvig.app.R
+import com.hedvig.app.feature.language.LanguageViewModel
+import com.hedvig.app.feature.settings.Language
+import com.hedvig.app.util.extensions.view.setHapticClickListener
+import com.hedvig.app.util.spring
+import kotlinx.android.synthetic.main.language_item_new.view.*
+
+class LanguageAdapterNew(
+    private val languageViewModel: LanguageViewModel,
+    private val selectedMarket: Market
+) : RecyclerView.Adapter<LanguageAdapterNew.ViewHolder>() {
+    private var lastChecked: RadioButton? = null
+    private var lastCheckedPos = 0
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(
+        LayoutInflater.from(parent.context)
+            .inflate(R.layout.language_item_new, parent, false)
+    )
+
+    override fun getItemCount() = 2
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        languageViewModel.selectedLanguage.postValue(null)
+        when (position) {
+            LOCAL -> {
+                when (selectedMarket) {
+                    Market.SE -> {
+                        holder.apply {
+                            language.text =
+                                language.resources.getString(R.string.SETTINGS_LANGUAGE_SWEDISH)
+                            parent.setHapticClickListener {
+                                languageViewModel.selectLanguage(Language.SV_SE)
+                            }
+                        }
+                    }
+                    Market.NO -> {
+                        holder.apply {
+                            language.text = language.resources.getString(R.string.norwegian)
+                            parent.setHapticClickListener {
+                                languageViewModel.selectLanguage(Language.NB_NO)
+
+                            }
+                        }
+                    }
+                }
+            }
+            EN -> holder.apply {
+                language.text =
+                    language.resources.getString(R.string.SETTINGS_LANGUAGE_ENGLISH)
+            }
+        }
+
+        if (position == 0 && holder.button.isChecked) {
+            lastChecked = holder.button
+            lastCheckedPos = 0
+        }
+
+        holder.parent.setHapticClickListener { v ->
+            when (position) {
+                EN -> languageViewModel.selectLanguage(Language.EN_SE)
```

Comment:
Don't we need to set language to `en_NO` when we've selected the Norwegian market? 🙂